server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -207,7 +208,7 @@ public class Manager extends AbstractServer
 
   private ManagerState state = ManagerState.INITIAL;
 
-  Fate<Manager> fate;
+  private final AtomicReference<Fate<Manager>> fateReady = new AtomicReference<>(null);
 
```

Comment:
```suggestion
private final CountDownLatch fateLatch = new CountDownLatch(1);
```

Reply:
I considered that, but having the fate object and the countdownlatch in separate variables means that separate threads wouldn't necessarily see the updated fate object unless I also made that voloatile, which seemed redundant. The best way to do it seemed to have the volatile reference be the same object that is monitored for readiness.

Reply:
On second thought, I suppose I could still do this. It would save me from looping.

Reply:
I was suggesting using both:
```
private final AtomicReference<Fate<Manager>> fateReady = new AtomicReference<>(null);
private final CountDownLatch fateLatch = new CountDownLatch(1);
```

You don't need to make `fateReady` volatile, as AtomicReference.get() and set() already have the memory semantics of volatile.

Reply:
> On second thought, I suppose I could still do this. It would save me from looping.

Yes, that's the reason I suggested it. I should have been more clear.

Reply:
Done in f9aa22b54853b5726945870ed575317f570843cf